app/src/main/java/app/grapheneos/camera/capturer/ImageSaver.kt

```diff
@@ -324,6 +332,7 @@ class ImageSaver(
 
     // implementation of ImageCapture.OnImageCapturedCallback.onError
     override fun onError(exception: ImageCaptureException) {
+        if (isCancelled) return
```

Comment:
It's not thread-safe to read value of `isCancelled` in `onError` or in `onCaptureSuccess()`. These callbacks are not executed on the main thread.

Reply:
Using an AtomicBoolean doesn't resolve this race condition fully. Main thread callback can be scheduled already by the time `isCancelled` is set. Cancellation check should happen on the main thread.

Reply:
Have updated the implementation to ensure both reading and writing happens on the same main thread. Thanks for looking into this!